books/data_transformation.py

```diff
@@ -1,108 +1,128 @@
+        users_frame['user_country'] = users_frame['user_country'].str.title()
+        users_frame['user_age'] = users_frame['user_age'].fillna(0).astype(int)
+        self.data_frames['users'] = users_frame
+
+    def transform_ratings(self, source_path):
+        ratings_frame = pd.read_csv(os.path.join(source_path, 'ratings.csv'), usecols=['ISBN', 'Book-Rating'], delimiter=';', encoding='windows-1250', low_memory=False)
+        new_ratings_column_names = {
+            'ISBN': 'r_book_isbn',
+            'Book-Rating': 'r_book_rating',
+        }
+        ratings_frame.rename(columns=new_ratings_column_names, inplace=True)
+        self.data_frames['ratings'] = ratings_frame
+
+    def transform_user_api_data(self, source_path):
+        with open(os.path.join(source_path, 'users_api_data.json'), 'r', encoding='utf-8') as f:
+            users_api_data = json.load(f)
+        users = users_api_data.get('results', [])
+        df_users = pd.json_normalize(users)
+        df_users.reset_index(inplace=True)
+        df_users = df_users.drop(columns=[
+            'index', 'gender', 'email', 'phone', 'cell', 'nat', 'name.title',
+            'name.first', 'name.last', 'location.street.number',
+            'location.street.name', 'location.state', 'location.postcode',
+            'location.coordinates.latitude', 'location.coordinates.longitude',
+            'location.timezone.offset', 'location.timezone.description',
+            'login.uuid', 'login.username', 'login.password', 'login.salt',
+            'login.md5', 'login.sha1', 'login.sha256', 'dob.date', 'dob.age',
+            'registered.date', 'id.name', 'id.value',
+            'picture.large', 'picture.medium', 'picture.thumbnail'
+        ])
+
+        new_api_users_column_names = {
+            'location.city': 'user_city',
+            'location.country': 'user_country',
+            'registered.age': 'user_age'
+        }
+        df_users.rename(columns=new_api_users_column_names, inplace=True)
+        df_users['user_age'] = df_users['user_age'].fillna(0).astype(int)
+        self.data_frames['api_users'] = df_users
+
+    def transform_scrapped_books_data(self, source_path):
+        scrapped_data = pd.read_csv(os.path.join(source_path, 'scrapped_book_data.csv'), delimiter=';', usecols=['Title', 'Author', 'ISBN/ISSN:', 'Wydawnictwo:', 'Rok wydania:'], low_memory=False)
+        new_scraped_data_column_names = {
+            'ISBN/ISSN:': 'book_isbn',
+            'Title': 'book_title',
+            'Author': 'book_author',
+            'Rok wydania:': 'year_published',
+            'Wydawnictwo:': 'publisher',
+        }
+        scrapped_data.rename(columns=new_scraped_data_column_names, inplace=True)
+        scrapped_data['year_published'] = scrapped_data['year_published'].fillna(0).astype(int)
+        self.data_frames['scrapped_data'] = scrapped_data
+
+    def save_to_csv(self, data_type, target_path):
```

Comment:
`data_type` is very missleading here :/

Reply:
Generally this method could be moved outside of a class as it shoudl only save a dataframe nothing else.